### PR TITLE
Add encoding start and end addresses config data command

### PR DESCRIPTION
**Pull request by microbit-carlos**
PR for the current proposal.

### CODE REVIEW

**Comment by finneyj** [comment]
Thanks @microbit-carlos.

Quick query: why BASE64 rather than just 2 HEX digits?  I know BASE64 is more efficient, but I though mapping each byte to two HEX digits would be simpler and more deterministic to compute in DAPLink?  

i.e. the output data will always be twice as long as the input data... so it's easy to calculate offsets etc when a BLOCK READ request arrives from the USB host?

**Comment by microbit-carlos** [comment]
Yeah, the encoding is still open, I just wrote something down to get the PR up (and one consideration for base64 is the hope that it can be decoded by some browser supported API, to reduce the HTML/JS size). There are more details in Jonny's email if you haven't seen it yet.

**Comment by gerargz** [comment]
Hi @microbit-carlos , I forgot about this repo and instead updated the spec in microbit-foundation/microbit-design@05a25eb. Can you review the changes? Let me know if you need me to raise a PR here with those changes.

File: spec/index.md [hunk 1]
```diff
@@ -273,8 +275,12 @@ KL27 storage memory layout:
     - default → DATA.BIN
   - File size (from start of data section)  (`0x02`)
     - 4B size in bytes (MSB First)
-    - default → 126 KBs
+    - default → 129024 (126 KBs)
     - max size is 126 KBs (128KB - 1KB for flash interface config and 1KB for DAL's config)
+  - Base64 encoding start address (from start of data section)  (`0x09`)
```
Comment (finneyj) [inline spec/index.md:280]: Is Base64 accurate here? i thought we were encoding each byte to two hex-digits rather then using Base64? if so, can we just change the description here?

File: spec/index.md [hunk 2]
```diff
@@ -244,9 +244,11 @@ KL27 storage memory layout:
 ↓ KL27 flash address 0x20000                                ↓ KL27 flash address0x40000
 ↓         ↓ KL27 flash address 0x20400                      ↓
 ┌---------┬-------------------------------------------------┐
-|[config] |[data                           [ DAL’s config ]]|
+| KL27    | storage data                                    |
+| config  |[ file.ext -------------------- ][ DAL’s config ]|
 └---------┴-------------------------------------------------┘
-          ↑ storage address 0x0000         ↑ storage address set by file size
+          ↑ storage address 0x0000          ↑ address set by file size
+                 ↑ base64 start address (anywhere inside file.ext range)
```
Comment (finneyj) [inline spec/index.md:251]: FYI: also mentions base64 here

**Comment by finneyj** [comment]
Looks fine to me other than checking the use of "base64" in the docs.

This will be enough to enable binary file system visualization. 

Query though: should we also have an end address for this encoding? So specify a "window" which to perform the encoding, rather than assume that the encoding should be done until the end of the file? 

I don't think we need it right now, but might be useful for future proofing? What do you think @microbit-carlos  @gerargz  ?

File: spec/index.md (hunk 1 above)
Comment (finneyj) [reply spec/index.md:280]: sorry @gerargz  - my bad! :)

**Comment by microbit-carlos** [comment]
Thanks everyone, I've update the the PRs with the comments.
@finneyj @gerargz could you have another look to ensure everything is correct?

File: spec/index.md [hunk 3]
```diff
@@ -273,14 +273,19 @@ KL27 storage memory layout:
...
     - max size is 126 KBs (128KB - 1KB for flash interface config and 1KB for DAL's config)
-  - Base64 encoding start address (from start of data section)  (`0x09`)
-    - 4B size in bytes (MSB First)
-    - default → 129024 (end of file address)
-    - max value is the end of file adress
+    - A remount is needed for the file size change to take effect
+  - Set encoding window (`0x09`)
+    - 4B Encoding window start (MSB First)
+    - 4B Encoding window end (MSB First)
+    - default → ????
```
[the remark this answers is not in the document]
Comment (microbit-carlos) [reply spec/index.md]: Updated, thanks Gerardo!

File: spec/index.md (hunk 2 above)
Comment (microbit-carlos) [reply spec/index.md]: Updated the diagram to include both pointers, thanks Joe!

File: spec/index.md (hunk 1 above)
Comment (microbit-carlos) [reply spec/index.md]: Updated with the content from microbit-foundation/microbit-design@05a25eb

**Comment by gerargz** [comment]
Looks good to me as well! 👍

**Comment by microbit-carlos** [comment]
Awesome, thanks everyone! I'll keep this open in case any tweaks are needed after implementation on the CODAL side.

**Pull request merged**